File: src/main/java/legends/models/TaskStatus.java
package legends.models;

public class TaskStatus {

	private final int taskCount;
	private final int totalTaskCount;
	private final int currentTaskID;

	public TaskStatus(int taskCount, int totalTaskCount, int currentTaskID) {
		this.taskCount = taskCount;
		this.totalTaskCount = totalTaskCount;
		this.currentTaskID = currentTaskID;
	}

	public boolean isFinished() {
		return taskCount == totalTaskCount;
	}

	public int getCurrentTaskID() {
		return currentTaskID;
	}
}

File: src/main/kotlin/legends/models/TaskStatus.kt
package legends.models

enum class TaskStatus {
    RUNNING,
    SUCCESS,
    FAIL,
    SKIP;

    companion object  {
        fun valueOfSafety(value: String): TaskStatus {
            return when(value.toUpperCase()) {
                RUNNING.name -> RUNNING
                SUCCESS.name -> SUCCESS
                FAIL.name -> FAIL
                SKIP.name -> SKIP
                else -> SKIP
            }
        }
    }
}